### PR TITLE
Add the `pytest` dependency for asv.

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Follow-up #4176 

Fix https://github.com/optuna/optuna/actions/runs/3690074016

## Description of the changes

This PR adds the `pytest` dependency for asv to fix the speed benchmark. It was broken at #4176.

https://github.com/optuna/optuna/blob/7cb75a060833e7d934fa7164c834d103288603d5/optuna/testing/storages.py#L13

### CODE REVIEW

**Comment by c-bata** [comment]
@not522 Could you review this PR if you have time? I'd say you can merge this PR after you approved.

**Review by not522 — APPROVED** [review]
LGTM

**Pull request merged**
